File: sdkwork-ai-sdk-swift/Sources/API/AudioApi.swift
import Foundation

public class AudioApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Create translation
    public func createTranslation(body: CreateTranslationRequest? = nil, params: [String: Any]? = nil) async throws -> TranscriptionResponse? {
        let response = try await client.post(ApiPaths.aiPath("/audio/translations"), body: body, params: params, headers: nil, contentType: "multipart/form-data")
        return response as? TranscriptionResponse
    }

    /// Create transcription
    public func createTranscription(body: CreateTranscriptionRequest? = nil, params: [String: Any]? = nil) async throws -> TranscriptionResponse? {
        let response = try await client.post(ApiPaths.aiPath("/audio/transcriptions"), body: body, params: params, headers: nil, contentType: "multipart/form-data")
        return response as? TranscriptionResponse
    }

    /// Create speech
    public func createSpeech(body: SpeechRequest) async throws -> String? {
        let response = try await client.post(ApiPaths.aiPath("/audio/speech"), body: body)
        return response as? String
    }
}

File: sdkwork-ai-sdk-csharp/Api/AudioApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class AudioApi
    {
        private readonly HttpClient _client;

        public AudioApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Create translation
        /// </summary>
        public async Task<TranscriptionResponse?> CreateTranslationAsync(CreateTranslationRequest? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<TranscriptionResponse>(ApiPaths.AiPath("/audio/translations"), body, query, null, "multipart/form-data");
        }

        /// <summary>
        /// Create transcription
        /// </summary>
        public async Task<TranscriptionResponse?> CreateTranscriptionAsync(CreateTranscriptionPostRequest? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<TranscriptionResponse>(ApiPaths.AiPath("/audio/transcriptions"), body, query, null, "multipart/form-data");
        }

        /// <summary>
        /// Create speech
        /// </summary>
        public async Task<string?> CreateSpeechAsync(SpeechRequest body)
        {
            return await _client.PostAsync<string>(ApiPaths.AiPath("/audio/speech"), body);
        }
    }
}
